15-screening_many_models.Rmd

```diff
@@ -1,13 +1,491 @@
+```{r setup-ch15, include=F, echo=F}
+library(details)
+```
+
 # Screening Many Models
 
 **Learning objectives:**
 
-- Jon will prefill these.
+-   Use the `{parsnip}` addin to create a set of **model specifications.**
+
+-   Create a **workflow set** with the `{workflowsets}` package.
+
+-   Describe the purposes of the **workflow set columns.**
+
+-   Create a **workflow set** with a `recipe` preprocessor.
+
+-   Create a **workflow set** with a `{dplyr}` selector preprocessor.
+
+-   **Tune** and **evaluate workflow sets.**
+
+-   Use `workflowsets::workflow_map()` to **tune all models** in a workflow set.
+
+-   Use convenience functions such as `workflowsets::rank_results()` to **examine** workflow set tuning results.
+
+-   **Visualize** workflow set tuning results.
+
+-   Use `workflowsets::workflow_map` with `{finetune}` to **efficiently screen models** using the **racing approach.**
+
+-   **Compare** the **results of the racing approach** to the **results of the full workflow set screening.**
+
+-   **Finalize the best model** from a workflow set.
+
+## Obligatory Setup
+
+Using the [2021 World Happiness Report](https://www.kaggle.com/ajaypalsinghlo/world-happiness-report-2021). Why?
+
+-   Small
+-   Interesting
+
+<details>
+
+<summary>
+
+How I felt reading this chapter with `concrete` from `{modeldata}`
+
+</summary>
+
+![](images/15-woody-stare.jpg)
+
+</details>
+
+```{r}
+library(tidyverse)
+library(tidymodels)
+theme_set(theme_minimal(base_size = 16))
+
+df <- 
+  here::here('data', 'world-happiness-report-2021.csv') %>%
+  read_csv() %>% 
+  janitor::clean_names()
+
+df %>% skimr::skim()
+```
+
+```{r}
+df_selected <-
+  df %>%
+  select(
+    ladder_score,
+    logged_gdp_per_capita,
+    social_support,
+    healthy_life_expectancy,
+    freedom_to_make_life_choices,
+    generosity,
+    perceptions_of_corruption
+  )
+
+cors <-
+  df_selected %>% 
+  select(where(is.numeric)) %>% 
+  corrr::correlate() %>% 
+  rename(col1 = rowname) %>% 
```

Comment:
Looks like it's a versioning issue for corrr::correlate() 
![image](https://user-images.githubusercontent.com/38083823/115730507-5b57e780-a354-11eb-8260-8166b88c4cb7.png)

```suggestion
  rename(col1 = term) %>% 
```

Reply:
https://github.com/tidymodels/corrr/issues/117#issuecomment-712460391